split JSON API production setup doc to separate documents (#15409)
digital-asset/daml: docs/source/json-api/production-setup/scaling-and-redundancy.rst

(if, for example, your monitoring indicates that the *HTTP JSON API* processing time is the bottleneck),
or have each HTTP JSON API instance talk to its own independent query store
(if the database response times are the bottleneck).

In the latter case, the Daml privacy model ensures that the *HTTP JSON API* requests
are made using the user-provided token, thus the data stored in a given
query store will be specific to the set of parties that have made queries through
that specific query store instance (for a given template).
Therefore, if you do run with separate query stores, it may be useful to route queries
(using a reverse proxy server) based on requesting party (and possibly queried template),
which would minimize the amount of data in each query store as well as the overall
redundancy of said data.

Users may consider running PostgreSQL backend in a `high availability configuration <https://www.postgresql.org/docs/current/high-availability.html>`__.
The benefits of this are use-case dependent as this may be more expensive for
smaller active contract datasets, where re-initializing the cache is cheap and fast.

> Finally we recommend using orchestration systems or load balancers which monitor the health of

Reviewer: ```suggestion
Finally, we recommend using orchestration systems or load balancers which monitor the health of
```

the service and perform subsequent operations to ensure availability. These systems can use the
`healthcheck endpoints <https://docs.daml.com/json-api/index.html#healthcheck-endpoints>`__
provided by the *HTTP JSON API* server. This can also be tied into supporting an arbitrary
autoscaling implementation in order to ensure a minimum number of *HTTP JSON API* servers on
failures.

> Set up the HTTP JSON API Service to work with Highly Available Participants

Reviewer: ```suggestion
Set Up the HTTP JSON API Service To Work With Highly Available Participants
```

***************************************************************************

> In case the participant node itself is configured to be highly available, depending on the setup you may want

Reviewer: ```suggestion
If the participant node itself is configured to be highly available, depending on the setup you may want
```

> to choose different approaches to connect to the participant nodes. In most setups, including those based on Canton,

Reviewer: ```suggestion
to choose different approaches to connect to the passive participant node(s). In most setups, including those based on Canton,
```
Is this what you meant?
